From 01fcb98bc7fa9399752667a5b786c0e76f418795 Mon Sep 17 00:00:00 2001
From: Alexander Holler <holler@ahsoftware.de>
Date: Wed, 17 Feb 2010 00:07:47 +0100
Subject: [PATCH 13/14] DockStar: Workaround to get the kernel loaded after a reboot.

Rebooting here always failed with:

scanning bus for devices... 3 USB Device(s) found
       scanning bus for storage devices... Device NOT ready
   Request Sense returned 02 04 02

But I've found a workaround using CONFIG_EHCI_HCD_INIT_AFTER_RESET
and a reset if scanning failed. After the reset scanning works.

Fixing the problem would be better, but therefor I would have to dig into
the USB-stack/USB-driver of U-Boot.
---
 include/configs/sheevaplug.h |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index 42b873b..f89f795 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -143,7 +143,9 @@
 #define CONFIG_BOOTCOMMAND \
         "${x_bootcmd_usb}; "    \
         "setenv bootargs ${x_bootargs} ${x_bootargs_root}; "    \
-        "${x_bootload_kernel}; bootm 0x800000;"
+        "setenv x_startkernel 'bootm 0x800000';" \
+        "run x_bootload_kernel x_startkernel;" \
+        "reset" /* If loading from USB failed we just reset, my experience was that the second time the device will be found */
 
 #define CONFIG_MTDPARTS		"orion_nand:1m(u-boot),"	\
 	"4m@1m(uImage),32m@5m(root),219m@37m(data) rw\0"
@@ -198,6 +200,7 @@
 #ifdef CONFIG_CMD_USB
 #define CONFIG_USB_EHCI			/* Enable EHCI USB support */
 #define CONFIG_USB_EHCI_KIRKWOOD	/* on Kirkwood platform	*/
+#define CONFIG_EHCI_HCD_INIT_AFTER_RESET
 #define CONFIG_EHCI_IS_TDI
 #define CONFIG_USB_STORAGE
 #define CONFIG_DOS_PARTITION
-- 
1.6.4.4

